src/evo/specialtxman.cpp

```diff
@@ -50,11 +50,18 @@ static bool CheckSpecialTxInner(const CTransaction& tx, const CBlockIndex* pinde
             }
             return CheckMNHFTx(tx, pindexPrev, state);
         case TRANSACTION_ASSET_LOCK:
-        case TRANSACTION_ASSET_UNLOCK:
             if (!llmq::utils::IsV20Active(pindexPrev)) {
                 return state.Invalid(TxValidationResult::TX_CONSENSUS, "assetlocks-before-v20");
             }
             return CheckAssetLockUnlockTx(tx, pindexPrev, indexes, state);
+        case TRANSACTION_ASSET_UNLOCK:
+            if (Params().NetworkIDString() == CBaseChainParams::REGTEST && !llmq::utils::IsV20Active(pindexPrev)) {
+                return state.Invalid(TxValidationResult::TX_CONSENSUS, "assetunlocks-before-v20");
+            }
+            if (Params().NetworkIDString() != CBaseChainParams::REGTEST && !llmq::utils::IsMNRewardReallocationActive(pindexPrev)) {
+                return state.Invalid(TxValidationResult::TX_CONSENSUS, "assetunlocks-before-mn_rr");
+            }
```

Comment:
why can't we just do
```suggestion
            if (!llmq::utils::IsMNRewardReallocationActive(pindexPrev)) {
                return state.Invalid(TxValidationResult::TX_CONSENSUS, "assetunlocks-before-mn_rr");
            }
```
and adjust tests accordingly instead?

Reply:
> and adjust tests accordingly instead?

non-trivial change due to withdrawal limits.

Reply:
We probably should in future PRs; this is intended to just quickly resolve this.

Reply:
Completely agree; but here we're just changing which version activates it on regtest; not really actually changing any functionality. But I do completely agree this should be changed and fix tests before rc.2